api/envoy/extensions/filters/http/compressor/v3/compressor.proto

```diff
@@ -128,3 +129,24 @@ message Compressor {
   // The last compressor which enables choose_first will be chosen if multiple compressor filters in the chain have choose_first as true.
   bool choose_first = 9;
 }
+
+message ResponseDirectionOverrides {
+}
+
+// Per-route overrides. As per-route overrides are needed, they should be
+// added here, mirroring the structure of `Compressor`. All fields should be
+// optional, to allow overriding arbitrary subsets of configuration.
+message CompressorOverrides {
+  // If present, response compression is enabled.
+  ResponseDirectionOverrides response_direction_config = 1;
+}
+
+message CompressorPerRoute {
+  oneof override {
+    option (validate.required) = true;
+
+    bool disabled = 1 [(validate.rules).bool = {const: true}];
+
+    CompressorOverrides overrides = 2;
```

Comment:
Another option would be to use `Compressor` as the type of `overrides`, but it would require changing all of its scalar fields to `optional`. I think that would be ok, but there may be consequences I'm unaware of.

If we can do that, it would be nice to avoid having parallel config and override message hierarchies.

Reply:
@envoyproxy/api-shepherds regarding the use of optional.